conan/tools/gnu/pkgconfigdeps.py

```diff
@@ -416,20 +340,51 @@ def content(self):
+                self._conanfile.output.warning(
+                    "PkgConfigDeps.build_context_suffix attribute has been "
+                    "deprecated. Use PkgConfigDeps.build_context_folder instead."
+                )
             # Check if it exists both as require and as build require without a suffix
-            self._validate_build_requires(host_req, build_req)
+            activated_br = {r.ref.name for r in build_req.values()
+                            if r.ref.name in self.build_context_activated}
+            common_names = {r.ref.name for r in host_req.values()}.intersection(activated_br)
+            without_suffixes = [common_name for common_name in common_names
+                                if not self.build_context_suffix.get(common_name)]
+            if without_suffixes:
+                raise ConanException(
+                    f"The packages {without_suffixes} exist both as 'require' and as"
+                    f" 'build require'. You need to specify a suffix using the "
+                    f"'build_context_suffix' attribute at the PkgConfigDeps generator.")
         elif self.build_context_folder is not None and self.build_context_suffix:
-            raise ConanException("It's not allowed to define both PkgConfigDeps.build_context_folder "
-                                 "and PkgConfigDeps.build_context_suffix (deprecated).")
+            raise ConanException(
+                "It's not allowed to define both PkgConfigDeps.build_context_folder "
+                "and PkgConfigDeps.build_context_suffix (deprecated).")
 
         for require, dep in list(host_req.items()) + list(build_req.items()) + list(test_req.items()):
             # Filter the build_requires not activated with PkgConfigDeps.build_context_activated
             if require.build and dep.ref.name not in self.build_context_activated:
                 continue
+            yield require, dep
+
+    @property
+    def content(self):
+        """
+        Get all the .pc files content
+        """
+        def _pc_file_name(name_, is_build_context=False, has_suffix=False):
+            # If no suffix is defined, we can save the *.pc file in the build_context_folder
+            build = is_build_context and self.build_context_folder and not has_suffix
+            # Issue: https://github.com/conan-io/conan/issues/12342
+            # Issue: https://github.com/conan-io/conan/issues/14935
+            return f"{self.build_context_folder}/{name_}.pc" if build else f"{name_}.pc"
+
+        pc_files = {}
+        for require, dep in self.get_dependencies():
+            suffix = self.build_context_suffix.get(require.ref.name, "") if require.build else ""
             # Save all the *.pc files and their contents
-            pc_files.update(_PCGenerator(self, require, dep).pc_files)
+            for name, content in _PCFilesDeps(self, require, dep, suffix=suffix).generate().items():
```

Comment:
If _PCFilesDeps is internal, you can implement directly its ``items()`` method, not necessary to have its own ``generate()`` (that doesn't save to disk) then ``items()``